tools/run_tests/xds_k8s_test_driver/framework/infrastructure/gcp/network_services.py

```diff
@@ -22,9 +23,134 @@
 
 logger = logging.getLogger(__name__)
 
+_ComputeV1 = gcp.compute.ComputeV1
+GcpResource = _ComputeV1.GcpResource
+
 
 class NetworkServicesV1Alpha1(gcp.api.GcpStandardCloudApiResource):
     ENDPOINT_CONFIG_SELECTORS = 'endpointConfigSelectors'
+    GRPC_ROUTES = 'grpcRoutes'
+    ROUTERS = 'routers'
+
+    @dataclasses.dataclass(frozen=True)
+    class Router:
+
+        name: str
+        url: str
+        type: str
+        network: Optional[str]
+        routes: Optional[List[str]]
+
+        @staticmethod
+        def from_dict(name: str, d: dict) -> 'Router':
+            return NetworkServicesV1Alpha1.Router(
+                name=name,
+                url=d["name"],
+                type=d["type"],
+                network=d.get("network"),
+                routes=list(d["routes"]) if "routes" in d else None,
+            )
+
+    @dataclasses.dataclass(frozen=True)
+    class GrpcRoute:
+
+        @dataclasses.dataclass(frozen=True)
+        class MethodMatch:
+            type: Optional[str]
+            grpc_service: Optional[str]
+            grpc_method: Optional[str]
+            case_sensitive: Optional[bool]
+
+            @staticmethod
+            def from_dict(d: dict) -> 'MethodMatch':
+                return MethodMatch(
+                    type=d.get("type"),
+                    grpc_service=d.get("grpcService"),
+                    grpc_method=d.get("grpcMethod"),
+                    case_sensitive=d.get("caseSensitive"),
+                )
+
+        @dataclasses.dataclass(frozen=True)
+        class HeaderMatch:
+            type: Optional[str]
+            key: str
+            value: str
+
+            @staticmethod
+            def from_dict(d: dict) -> 'HeaderMatch':
```

Comment:
Note that the return type, even when in quotes, should be resolvable in the current context:
![image](https://user-images.githubusercontent.com/672669/127395192-b43188ec-9a82-4405-8f20-6c0e20a278e1.png)

So this should be `NetworkServicesV1Alpha1.GrpcRoute.HeaderMatch`.

I know how annoying this may be to repeat `NetworkServicesV1Alpha1` everywhere.  
Feel free to move these dataclasses to the top level, I ended up doing the same in the lates IAM API:
https://github.com/grpc/grpc/blob/c472a2a58c18e08f8ae8e6710cb071eaf631e98c/tools/run_tests/xds_k8s_test_driver/framework/infrastructure/gcp/iam.py#L59-L66